### ISSUE TITLE
`ignorePatterns` glob is no longer relative to the project root

### ISSUE THREAD

**Issue by DetachHead** [microsoft/vscode-mypy#338]
this seems to be a regression caused by #220

the `ignorePatterns` used to work relative to the root of the project, but this no longer seems to be the case. for example when ignoring the file `foo.py` in the project root, the pattern `foo.py` used to work, but now i have to use `**/foo.py` because it appears to now be matching against the full path instead.

## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): 3.13.1
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): uv
- Operating system (and version): windows 11
- Version of tool extension you are using: 2024.0.0

### Expected Behavior
file is ignored

### Actual Behavior
file is not ignored

## Reproduction Steps:
specify a file that's located in the project root to be ignored, eg:
```json
{
    "mypy-type-checker.ignorePatterns": ["foo.py"]
}
```
```py
# ./foo.py
foo: int = ""
```
## Logs:

<details>
  <summary>Click here for detailed logs</summary>

```
2025-01-09 11:12:32.073 [info] c:\Users\user\project\.venv\Scripts\python.exe -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --ide --hide-error-code-links --show-error-end c:\Users\user\project\foo.py
2025-01-09 11:12:32.073 [info] [Trace - 11:12:32 AM] Received notification 'window/logMessage'.
2025-01-09 11:12:32.073 [info] CWD Server: c:\Users\user\project
2025-01-09 11:12:34.796 [info] [Trace - 11:12:34 AM] Received notification 'window/logMessage'.
2025-01-09 11:12:34.797 [info] file:///c%3A/Users/user/project/foo.py :
c:\Users\user\project\foo.py:1:12:1:13: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]

2025-01-09 11:12:34.797 [info] [Trace - 11:12:34 AM] Received notification 'window/logMessage'.
2025-01-09 11:12:34.797 [info] file:///c%3A/Users/user/project/foo.py :
c:\Users\user\project\foo.py:1:12:1:13: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]

2025-01-09 11:12:34.797 [info] [Trace - 11:12:34 AM] Received notification 'textDocument/publishDiagnostics'.
```

</details>
